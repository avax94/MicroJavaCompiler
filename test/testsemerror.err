2:10 Tip konstante i konstanta se ne poklapaju!
3:11 Tip konstante i konstanta se ne poklapaju!
12:4 Povratni tip metode i naredbe return se ne poklapaju
16:4 Ne postoji simbol this u tabeli simbola!
17:2 Nedostaje return statement u metodi!
21:9 Klasa mora nasledjivati drugu klasu!
28:2 Kljucna rec static se moze koristiti samo uokviru klase!
28:13 Globalne metode ne mogu biti staticke!
32:4 Izraz je pogresnog tipa!
33:4 Operacija se moze izvesti samo na tipu: int!
34:10 Parametri operanda new moraju biti tipa int!
35:4 Izraz je pogresnog tipa!
36:4 break se sme koristiti samo uokviru petlje!
37:4 continue se sme koristiti samo uokviru petlje!
38:4 Ne moze se pozvati READ za ovaj tip simbola!
39:4 Ne moze se pozvati PRINT za ovaj tip simbola!
40:14 Sabirci moraju biti tipa int!
40:4 Izraz je pogresnog tipa!
42:7 Ne moze se koristiti ovaj operand sa klasama!
45:9 45:9 Simbol nije klasnog tipa
46:4 Izraz je pogresnog tipa!
47:4 Promenljival1 nije klasnog tipa!
48:4 Simbol l1 nije metoda!
50:7 Pogresan tip 0. argumenta!
50:12 Pogresan tip 1. argumenta!
