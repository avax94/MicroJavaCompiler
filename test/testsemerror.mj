program syntest
const int k1 = 'c';
const char c1 = 2;
int x, y, z;

class C1 {
int f1, f2;
char f3;
C1 c;
{
  int m1(int p1, char p2) {
    return p2;
  }

  static int s1() {
    this.c = c;
  }
}
}

class C2 extends int {
{

}
}

{
  static int f1(int x, int y) {
     return 0;
  }
  void main() int l1, l2, l3; C1 c1; C2 c2; C1 niz[]; {
    niz = l1 + l2;
    c1++;
    niz = new C1['c'];
    l1 = 'c';
    break;
    continue;
    read(c2);
    print(c1);
    l1 = c1 + c2;

    if(c1 < c2) {

    }
    l3 = new int;
    c1+= 2;
    l1.l1 = l2;
    l1(l1, l2);

    f1('c', 'd');

   }
}

