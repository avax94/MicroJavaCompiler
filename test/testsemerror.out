=====================SEMANTICKA ANALIZA=========================
Pretraga na (12), nadjeno Var p2: char, 2, 1 
Pretraga na (16), nadjeno Fld c: Class, 4, 1 
Pretraga na (32), nadjeno Var niz: Arr of Class, 5, 1 
Pretraga na (32), nadjeno Var l1: int, 0, 1 
Pretraga na (32), nadjeno Var l2: int, 1, 1 
Pretraga na (33), nadjeno Var c1: Class, 3, 1 
Pretraga na (34), nadjeno Var niz: Arr of Class, 5, 1 
Pretraga na (35), nadjeno Var l1: int, 0, 1 
Pretraga na (38), nadjeno Var c2: Class, 4, 1 
Pretraga na (39), nadjeno Var c1: Class, 3, 1 
Pretraga na (40), nadjeno Var l1: int, 0, 1 
Pretraga na (40), nadjeno Var c1: Class, 3, 1 
Pretraga na (40), nadjeno Var c2: Class, 4, 1 
Pretraga na (42), nadjeno Var c1: Class, 3, 1 
Pretraga na (42), nadjeno Var c2: Class, 4, 1 
Pretraga na (45), nadjeno Var l3: int, 2, 1 
Pretraga na (46), nadjeno Var c1: Class, 3, 1 
Pretraga na (47), nadjeno Var l1: int, 0, 1 
Pretraga na (47), nadjeno Var l2: int, 1, 1 
Pretraga na (48), nadjeno Var l1: int, 0, 1 
Pretraga na (48), nadjeno Var l1: int, 0, 1 
Pretraga na (48), nadjeno Var l2: int, 1, 1 
Pretraga na (50), nadjeno Meth f1: int, 18, 2 
=====================SINTAKSNA ANALIZA=========================
Pronadjeno: 3 globalnih promenljivih
Pronadjeno: 2 globalnih konstanti
Pronadjeno: 0 globalnih nizova
Pronadjeno: 6 lokalnih promenljivih
Pronadjeno: 3 globalnih funkcija
Pronadjeno: 2 definicija klasa
Pronadjeno: 1 metoda
Pronadjeno: 1 statement blokova
Pronadjeno: 4 poziva funkcija u mainu
Pronadjeno: 4 formalnih argumenata
Pronadjeno: 4 polja klasa
=====================SYMBOL TABLE DUMP=========================
Var $helper: int, 0, 0 
Prog syntest: notype, 0, 0 

Con k1: int, 99, 0 
Con c1: char, 2, 0 
Var x: int, 1, 0 
Var y: int, 2, 0 
Var z: int, 3, 0 
Type C1: Class [Fld $vtable: int, 0, 1 Fld f1: int, 1, 1 Fld f2: int, 2, 1 Fld f3: char, 3, 1 Fld c: Class [], 4, 1 Meth m1: int, 0, 3 
   Var this: , 0, 1 
   Var p1: int, 1, 1 
   Var p2: char, 2, 1 
Meth s1: int, 8, 0 
], 0, 0 
Type C2: Class [Fld $vtable: int, 0, 1 ], 0, 0 
Meth f1: int, 18, 2 
   Var x: int, 0, 1 
   Var y: int, 1, 1 

Meth main: notype, 26, 0 
   Var l1: int, 0, 1 
   Var l2: int, 1, 1 
   Var l3: int, 2, 1 
   Var c1: Class [], 3, 1 
   Var c2: Class [], 4, 1 
   Var niz: Arr of Class, 5, 1 

Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 

